components/brave_wallet/common/eth_sign_typed_data_helper_unittest.cc

```diff
@@ -27,28 +27,62 @@ TEST(EthSignedTypedDataHelperUnitTest, EncodeTypes) {
         {"name": "wallet", "type": "address"}
     ]})");
 
+  const std::string invalid_types_json(R"({
+    "Domain": [
+        { "name": "name", "type": 1234 },
+        { "name": ["AStringArray", "String2"], "type": 1234 },
+        { "name": 1234, "type": "number" },
+        { "name": { "name": "name" }, "type": "object" },
+        { "name": "version", "type": "string" },
```

Comment:
We shouldn't mix bad case of name and type in one type, we need one type for bad name otherwise we always hit bad type first. Also we should move `invalid_types_json` and `invalid_types_value` to be in front of `invalid_types_value_helper` and just put all of bad case stuffs after all valid cases

Reply:
> We shouldn't mix bad case of name and type in one type, we need one type for bad name otherwise we always hit bad type first.

Oh good catch. I'll update that.

> Also we should move invalid_types_json and invalid_types_value to be in front of invalid_types_value_helper and just put all of bad case stuffs after all valid cases

Noted, thanks for clarifying this pattern. I wasn't certain about the convention we use for this because I've always previously isolated each case in independent unit tests. I noticed that we tend to combine failure and success cases into the same unit test presumably to make the tests run faster.

I appreciate your insightful feedback here as I start trying to learn the codebase more.

Reply:
Should be all fixed in this [2c716e6](https://github.com/brave/brave-core/pull/16883/commits/2c716e6418ed7df75b9b06c1b474ffb25186bc44)

I attempted to isolate these tests out a bit further, but wasn't sure if that's actually what you were after so if you see a way to combine these to be a bit more efficient happy to do so. In terms of performance tradeoffs the unit test is still sub 0ms so I figured it wouldn't matter too much if I over isolated.

Reply:
Since the failure case is now expanded, we should move it to separate test suite, it was ok to combine success and failure into one test suite when it is small.  Please use for loop to reduce the redundant code and you can grep `SCOPED_TRACE` which will be printed when a case failed our expectation so we will know which is the one.

Reply:
I've reduced the failure cases substantially by removing the redundant code (and unused JSON properties) in [d6e2623](https://github.com/brave/brave-core/pull/16883/commits/d6e2623bc2acb491201e858e51a95f3d0827d64a). Even those there's a substantially smaller amount of code for the failure cases now I still split it into it's own test suite because I wanted force myself to figure out why the naming convention I was using (e.g. not matching the function name) wasn't working. I figured it out with the use of the `FRIEND_TEST_ALL_PREFIXES` macro, but I'm not sure what it's doing based on my read of the docs. Do you have a good high level explanation of what's going on?

Reply:
`EncodeTypes` you use is a private function, that is why we need to friend the suite to use it.

Reply:
Ahh ok, that makes sense now. The name matching the private function seems irrelevant to whether it's accessible. I was thinking that had some effect on the ability to discover private functions, but it seems specific to this macro.